src/parameters.F90

```diff
@@ -2159,28 +2165,26 @@ subroutine param_read()
       call io_error('param_read: Guiding centres requested, but no projection block found')
     ! check to see that there are no unrecognised keywords
     if (found) then
-      ! if (scdm_proj) then
-      !   call io_error('param_read: Can not specify the projection block and scdm_proj=true at the same time.')
-      ! else
       call param_get_projections
-      ! end if
     end if
 
+    ! Constrained centres
     call param_get_block_length('slwf_centres', found, i_temp)
-    if (found) then
-      if (selective_loc .and. slwf_constrain .and. allocated(proj_site)) then
-        if (scdm_proj) call io_error('Error: Selective localisation with constrains is not compatible with SCDM.')
-        ! Centre constraints block
-        call param_get_centre_constraints
-      else
-        call io_error('Error: <slwf_centres> block found, but either no slwf_num or &
-             & slwf_constrain=false or &
-             & no projection_block specified.')
-      end if
+    if (found .and. .not. slwf_constrain) then
+      write (stdout, '(a)') ' slwf_constrain set to false. Ignoring <slwf_centres> block '
+      ! Check that either projections or constrained centres are specified if slwf_constrain=.true.
+    elseif (slwf_constrain .and. .not. (found .or. allocated(proj_site))) then
+      call io_error('Error: slwf_constrain = true, but neither &
+           & <slwf_centre> block  nor &
+           & <projection_block> are specified.')
     else
-      if (selective_loc .and. slwf_constrain .and. allocated(proj_site)) &
-        write (stdout, '(a)') 'No slwf_centres block requested. Desired centres for SLWF same as projection centres'
+      ! Allocate array for constrained centres
+      call param_get_centre_constraints
```

Comment:
Shouldn't there be a check that `selective_loc` is true? The function internally accesses variables that are now allocated only `if (selected_loc)` above

Reply:
Hi @giovannipizzi, I changed it so it is easier to follow. I don't think I need a check on `selective_loc`, since `slwf_constrain` is true only if `selective_loc` is also true.